app/models/email.rb

```diff
@@ -63,4 +59,11 @@ def self.email_available_for_user?(user, address)
   def self.send_confirmation(email, community)
     MailCarrier.deliver_later(PersonMailer.email_confirmation(email, community))
   end
+
+  def self.find_by_address_and_community_id(address, community_id)
+    Email
+      .joins(:community_memberships)
```

Comment:
I guess this will not work without explicitly defining the foreign key (community_memberships doesn't have `email_id`)